WMFData/Sources/WMFData/Data Controllers/Activity Tab/WMFActivityTabDataController.swift

```diff
@@ -185,6 +188,177 @@ public final class WMFActivityTabDataController {
         guard let project = WMFProject(id: page.projectID) else { return nil }
         return try await articleSummaryController.fetchArticleSummary(project: project, title: page.title)
     }
+
+    // MARK: - Experiment
+
+    @objc public func assignmentIntBlocking() -> Int {
+        let semaphore = DispatchSemaphore(value: 0)
+        var result = 0
+
+        Task.detached { [weak self] in
+            defer { semaphore.signal() }
+            guard let self else { return }
+
+            do {
+                let assignment = try await self.assignOrFetchExperimentAssignment()
+                if assignment == .activityTab {
+                    result = 1
+                } else {
+                    result = 0
+                }
+            } catch {
+                result = 0
+            }
+        }
+
+        semaphore.wait()
+        return result
+    }
+
+    public func assignOrFetchExperimentAssignment() async throws -> ActivityTabExperimentAssignment? {
```

Comment:
Does this need to be async? I don't see any awaits. If not you may be able to remove `assignmentIntBlocking`.